### PR TITLE
Fetch Professors and add to Section Popup 

### PR DESCRIPTION
**Pull request by jaskirat-gill**
In reference to #36 

- Fetches instructor information in util.ts from same HTML Element as Title & Section Details are.
- Add instructor name to ISectionData Interface
- Display instructor in SectionPopup below details
- Defaults to "No instructor Information" if not available

image

### CODE REVIEW

**Comment by l-currie** [comment]
Hello, can you please try this course: "CHEM_V 315-LC1 " and see if the instructors are working? It doesn't seem to be fetching properly on my end, possibly due to having multiple instructors.  
image

File: src/content/SectionPopup/SectionPopup.tsx [hunk 1]
```diff
@@ -30,6 +30,8 @@ const SectionPopup = ({
         <div className="SectionPopupTitle">{selectedSection?.code}</div>
         <hr />
         <div className="SectionPopupDetails">{selectedSection?.name}</div>
+        <div className="SectionPopupInstructor">Instructor: </div>
+        <div>{selectedSection?.instructor || "No Instructor Information"}</div>
```
Comment (StashC) [inline src/content/SectionPopup/SectionPopup.tsx:34]: Can we change the message to "unavailable" to keep it consistent with the grades information

**Comment by l-currie** [comment]
Also, it looks like in the screenshot you provided, the bottom buttons got pushed outside the popup background, I think we need to modify the CSS to fit everything, or have it be scrollable. @mlool scrollable might be better considering multiple instructors/adding more information. Thoughts?

File: src/content/SectionPopup/SectionPopup.tsx (hunk 1 above)
Comment (jaskirat-gill) [reply src/content/SectionPopup/SectionPopup.tsx]: Done.

**Comment by mlool** [comment]
> o

Yes, let's do scrollable for now. We can update the styling and rethink how the popup works later

**Comment by StashC** [comment]
As for the multiple instructors issue, we might be able to a) find the parent class which contains all the instructors, we can do this by finding the div which has the child with the data automation id 'compositeDetailTitle' and the word "Instructors".
Then we can go through all of the prompt option classes there and add them.
image

**Comment by jaskirat-gill** [comment]
Hey, really good point about multiple instructors. Looking into the courseLabel object, it seems like when there are multiple instructors that information is just not in the div where all the rest of the information is for some reason. The first console log is for CHEM 315 with multiple instructors and the second is for CPSC 320 which has 1 instructor. Both have atleast 2 labels (title and section details) but all classes with multiple instructors lack labels with instructors in this div. I'll look into fixing that

image

> Hello, can you please try this course: "CHEM_V 315-LC1 " and see if the instructors are working? It doesn't seem to be fetching properly on my end, possibly due to having multiple instructors. image

**Comment by jaskirat-gill** [comment]
@l-currie Fixed the issue with multiple instructors. 
The code now iterates through the labels with the promptOption data-automation-id, from my understanding and testing all the labels after the first 3 are either meeting times or professor names. Meeting times include the substring "|" but professor names do not. 
Set the popup to be scrollable, did not mess with any styling of the scrollbar for the sake of consistency with existing UI so that can be changed if you guys want

One issue I did observe is that in some cases the instructor information has not been loaded (as in the "Instructors" label does not exist) until the section box in workday is expanded and thus it shows as unavailable because when the HTML is fetched, instructor information does not exist. Don't know if this is something that we can fix, but will look into it



image

**Comment by jaskirat-gill** [comment]
Ignore above comment about it sometimes not loading. Turns out the DOM updating code written for the section details was enough, moved the code below it and it fixed that

File: src/content/utils.ts [hunk 2]
```diff
@@ -127,6 +126,16 @@ export async function extractSection(element: Element) {
   // Splitting title and section details into parts
   const titleParts = title.split(' - ');
 
+  // Concatenate section details back together if it got split 
+  if (titleParts.length > 2) {
```
Comment (StashC) [inline src/content/utils.ts:130]: Can we change this to just split on the first occurrence of " - " instead?  This way we don't have to go and recombine it and I think we can even remove the check on length > 2.

**Comment by StashC** [comment]
Sorry I probably won't get a chance to fully test this until Wednesday morning

**Comment by jaskirat-gill** [comment]
No worries. This PR is closed because the branch got deleted. The changes here should have been two seperate PRs but that has been fixed now with #67 and #68. Ignore this PR please

**Pull request closed without merging**